docs/recipes.rst

```diff
@@ -0,0 +1,51 @@
+.. _recipes:
+
+PEX Recipes and Notes
+=====================
+
+This is a space for info beyond the purpose or building of PEX files.
+
+Gunicorn and PEX
+----------------
+
+Normally, to run a wsgi-compatible application with Gunicorn, you'd just
+point Gunicorn at your application, tell Gunicorn how to run it, and you're
+ready to go - but if your application is shipping as a PEX file, you'll have
+to bundle Gunicorn as a dependency and set Gunicorn as your entry point. Gunicorn
+can't enter a PEX file to retrieve the wsgi instance, but that doesn't prevent
+the PEX from invoking Gunicorn.
+
+This retains the benefit of zero `pip install`'s to run your service, but it
+requires a bit more setup. First, ensure Gunicorn is packaged as a depenency, 
+the following assumes Flask as the wsgi framework, Django setup should be 
+similar:
+
+.. code-block:: bash
+
+    $ pex flask gunicorn myapp -e gunicorn.app.wsgiapp -o ~/service.pex
+    
+The dependency on Gunicorn can live in your setup.py as well:
```

Comment:
Although true, if the app has 0 lines of code that import from gunicorn this is not a clean recommendation. Consider dropping.

Reply:
Fair point, it is a bit of a hack. I added the option as it's a cleaner approach for build systems. In terms of code, the app definitely doesn't rely on Gunicorn - especially with Flask included as well - but from an ops view Gunicorn is required for the app to go live, and it's easier to have a standard PEX invocation at compile time that leans more on setup.py than to have special build configuration for each wsgi app.

Would it be more appropriate to shorten that bit to just line 27, and instead of a code block, have a caveat that it may make CI/CD easier but is not recommended when the app isn't using Gunicorn imports?

Reply:
Also a fair point, but I think at the wrong scope for these docs. I'd leave it out. You've already taught the reader how to use _pex_ to create a self-contained pex with gunicorn / wsgi container included.

If they want to use a pants BUILD file with a `python_binary` definition with the extra deps, write a side-car setup.py, a setup.py that depends on the original, or even modify the original setup.py except use [`extras_require`](https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-extras-optional-features-with-their-own-dependencies) instead of `install_requires` (say with an extra requirement list named 'deployable': `pex myapp[deployable] -e gunicorn.app.wsgiapp -o ~/service.pex), then they can probably figure that out.

Reply:
Great points, I dig it - redacted that section